src/doc/trpl/README.md

```diff
@@ -1,39 +1,221 @@
 % The Rust Programming Language
 
-Welcome! This book will teach you about [the Rust Programming
-Language](http://www.rust-lang.org/). Rust is a modern systems programming
-language focusing on safety and speed. It accomplishes these goals by being
-memory safe without using garbage collection.
+Welcome! This book will teach you about the [Rust Programming Language][rust].
+Rust is a systems programming language focused on three goals: safety, speed,
+and concurrency. It maintains these goals without having a garbage collector,
+making it a useful language for a number of use cases other languages aren’t
+good at: embedding in other languages, programs with specific space and time
+requirements, and writing low-level code, like device drivers and operating
+systems. It improves on current languages targeting this space by having a
+number of compile-time safety checks that produce no runtime overhead, while
+eliminating all data races. These checks can be used to form ‘zero-cost
+abstractions’, making Rust _feel_ like a high-level language, while giving you
```

Comment:
The line starting with "These checks" should not follow from the previous one; tt doesn't feel related. I feel the following is more fitting:

    Rust also aims to achieve zero-cost abstrations...

Also, them being zero-cost have nothing to do with them feeling high-level. These are separate things.

I am here assuming zero-cost refers to performance.